AK/FloatingPointStringConversions.cpp

```diff
@@ -0,0 +1,2018 @@
+            } else {
+                reparse_ptr = start_of_fractional_part;
+                char const* fractional_end = result.fractional_part.characters_without_null_termination() + result.fractional_part.length();
+
+                while (mantissa < smallest_eleven_digit_number && (fractional_end - reparse_ptr) >= 8) {
+                    mantissa = 100'000'000 * mantissa + eight_digits_to_value(read_eight_digits(reparse_ptr));
+                    reparse_ptr += 8;
+                }
+
+                while (mantissa < smallest_nineteen_digit_number && reparse_ptr != fractional_end) {
+                    mantissa = 10 * mantissa + (*reparse_ptr - '0');
+                    ++reparse_ptr;
+                }
+
+                // Again we might be truncating fractional number so scale the exponent with that
+                // However here need to subtract 1 from the exponent for every fractional digit
+                exponent = explicit_exponent - (reparse_ptr - start_of_fractional_part);
+            }
+        }
+    }
+
+    result.mantissa = mantissa;
+    result.exponent = exponent;
+    return result;
+}
+
+constexpr static u128 compute_power_of_five(i64 exponent)
+{
+    constexpr u4096 bit128 = u4096 { 1u } << 127u;
+    constexpr u4096 bit129 = u4096 { 1u } << 128u;
+
+    VERIFY(exponent <= 308);
+    VERIFY(exponent >= -342);
+
+    if (exponent >= 0) {
+        u4096 base { 1u };
+        for (auto i = 0u; i < exponent; ++i) {
+            base *= 5u;
+        }
+
+        while (base < bit128)
+            base <<= 1u;
+        while (base >= bit129)
+            base >>= 1u;
+
+        return u128 { base };
+    }
+
+    exponent *= -1;
+    if (exponent <= 27) {
+        u4096 base { 1u };
+        for (auto i = 0u; i < exponent; ++i) {
+            base *= 5u;
+        }
+
+        auto z = u4096::my_size() * 8
+            - base.clz();
+
+        auto b = z + 127;
+        u4096 base2 { 1u };
+        for (auto i = 0u; i < b; ++i) {
+            base2 *= 2u;
+        }
+
+        base2 /= base;
+        base2 += 1u;
+
+        return u128 { base2 };
+    }
+
+    VERIFY(exponent <= 342);
+    VERIFY(exponent >= 28);
+
+    u4096 base { 1u };
+    for (auto i = 0u; i < exponent; ++i) {
+        base *= 5u;
+    }
+
+    auto z = u4096::my_size() * 8
+        - base.clz();
+
+    auto b = 2 * z + 128;
+
+    u4096 base2 { 1u };
+    for (auto i = 0u; i < b; ++i) {
+        base2 *= 2u;
+    }
+
+    base2 /= base;
+    base2 += 1u;
+
+    while (base2 >= bit129)
+        base2 >>= 1u;
+
+    return u128 { base2 };
+}
+
+static constexpr i64 lowest_exponent = -342;
+static constexpr i64 highest_exponent = 308;
+
+constexpr auto pre_compute_table()
+{
+    // Computing this entire table at compile time is slow and hits constexpr
+    // limits, so we just compute a (the simplest) value to make sure the
+    // function is used. This means the code is stand alone in case we need to
+    // regenerate this table.
```

Comment:
How can it be regenerated?

Reply:
With the function above, I'll add that to the comment